sdcm/scan_operation_thread.py

```diff
@@ -0,0 +1,352 @@
+        3) Add a random CK filter with random row values.
+        :return: a CQL reversed-query
+        """
+        with self.create_session(self.db_node) as session:
+            ck_name = self.ck_name
+            rows_count = self.rows_count
+            ck_random_min_value = random.randint(a=1, b=rows_count)
+            ck_random_max_value = random.randint(a=ck_random_min_value, b=rows_count)
+            self.ck_filter = ck_filter = random.choice(list(self.reversed_query_filter_ck_by.keys()))
+            pk_name = self.pk_name
+            if pks := get_partition_keys(ks_cf=self.ks_cf, session=session, pk_name=pk_name):
+                partition_key = random.choice(pks)
+                # Form a random query out of all options, like:
+                # select * from scylla_bench.test where pk = 1234 and ck < 4721 and ck > 2549 order by ck desc
+                # limit 3467 bypass cache
+                normal_query = reversed_query = self.basic_query.format(
+                    self.ks_cf) + f' where {pk_name} = {partition_key}'
+                query_suffix = limit = ''
+
+                if random.choice([False] + [True]):  # Randomly add a LIMIT
+                    limit = random.randint(a=rows_count/5, b=rows_count)
+                    query_suffix += f' limit {limit}'
+                query_suffix = self.randomly_bypass_cache(cmd=query_suffix)
+
+                # Randomly add CK filtering ( less-than / greater-than / both / non-filter )
+
+                # example: rows-count = 20, ck > 10, ck < 15, limit = 3 ==> ck_range = [11..14] = 4
+                # ==> limit < ck_range
+                # reversed query is: select * from scylla_bench.test where pk = 1 and ck > 10
+                # order by ck desc limit 5
+                # normal query should be: select * from scylla_bench.test where pk = 1 and ck > 15 limit 5
+                if ck_filter == 'lt_and_gt':
+                    # Example: select * from scylla_bench.test where pk = 1 and ck > 10 and ck < 15 order by ck desc
+                    reversed_query += self.reversed_query_filter_ck_by[ck_filter].format(ck_name,
+                                                                                         ck_random_max_value,
+                                                                                         ck_name,
+                                                                                         ck_random_min_value)
+                    ck_range = ck_random_max_value - ck_random_min_value - 1  # e.g. 15 - 10 - 1 = 4
+                    if limit and limit < ck_range:
+                        # Example: select * from scylla_bench.test where pk = 1 and ck > 10 and ck < 15
+                        # order by ck desc limit 3
+                        # output of ck should be: [12,13,14]
+                        normal_query += \
+                            f' and {ck_name} < {ck_random_max_value} and {ck_name} >= {ck_random_max_value - limit}'
```

Comment:
I don't understand why this part is needed.
Why do you care here about the limit and how do the normal vs reversed matters?

Reply:
It should be because the normal query needs the inversed range. It is actually not really tested now since it requiers the completion of other opened task for that ("DeepDif").
Thus it will only be verified and tested to be correct when the other task is completed.
But the theoretical idea is yet as the example above.